config/platforms/platform_intelsgx.h.in

```diff
@@ -0,0 +1,74 @@
+#include <endian.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <mbusafecrt.h>
+
+/*
+ * Intel SGX enclaves are completely isolated from the underlying
+ * kernel and therefore syscalls are not allowed by default. This
+ * means that getting the current time and locale information is
+ * impossible unless there is intervention by the enclave developers.
+ *
+ * Thus enclave developers must define DUK_USE_DATE_GET_NOW and
+ * DUK_USE_DATE_GET_LOCAL_TZOFFSET for their enclave either manually
+ * in duk_config.h or via other means (other config header, compiler
+ * flags). If these are not defined compilation will fail with a
+ * preprocessor error.
+ *
+ * DUK_USE_DATE_GET_NOW should return a duk_double_t value containing
+ * the current ECMASCript time.
+ * DUK_USE_DATE_GET_LOCAL_TZOFFSET should return a duk_int_t value
+ * containing the offset from UTC in seconds of the current time.
+ * See the datetime.rst file for more information, as well as the
+ * duk_bi_date.c, duk_bi_date_unix.c files for inspiration.
+ *
+ * DUK_USE_DATE_FORMAT_STRING and DUK_USE_DATE_PARSE_STRING macros are
+ * not mandatory and are therefore not defined here, but enclave
+ * developers are free to define them via duk_config.h or other means
+ * if they are implementing platform or locale scpecific date parsing
+ * and formatting.
+ *
+ * All related Duktape provided implementations regarding date and
+ * time are intentionally disabled, since date and time functions are
+ * not generally available in Intel SGX enclaves.
+ */
+
+#undef DUK_USE_DATE_FMT_STRFTIME
+#undef DUK_USE_DATE_NOW_GETTIMEOFDAY
+#undef DUK_USE_DATE_NOW_TIME
+#undef DUK_USE_DATE_NOW_WINDOWS
+#undef DUK_USE_DATE_NOW_WINDOWS_SUBMS
+#undef DUK_USE_DATE_PRS_GETDATE
+#undef DUK_USE_DATE_PRS_STRPTIME
+#undef DUK_USE_DATE_TZO_GMTIME
+#undef DUK_USE_DATE_TZO_GMTIME_R
+#undef DUK_USE_DATE_TZO_GMTIME_S
+#undef DUK_USE_DATE_TZO_WINDOWS
+#undef DUK_USE_DATE_TZO_WINDOWS_NO_DST
+
+/*
+ * Intel SGX enclaves don't have access to sscanf, so pointer parsing must
+ * be without using sscanf.
+ */
+#define DUK_USE_MEMBASED_POINTER_ENCODING
```

Comment:
This probably won't work as expected: if this define appears here, it will prevent the emission of the default value *for all platforms* when duk_config.h default/forced options section is emitted. The problem is that the default/forced options section only knows that the define is forced in the preceding header, but it doesn't know (or care) that it is conditional to a platform.

So IMO this header should just say in its comments that `DUK_USE_MEMBASED_POINTER_ENCODING` is a required option for Intel SGX enclaves. There should be a better way to address this, but at present I don't think configure.py can do better.

One possible solution would be for platforms to optionally have a config check snippet at the end of `duk_config.h` where they could check that required options are met, so that they could `#error` out cleanly.